pkg/sync/kubernetes/featureflagconfiguration/clientset.go

```diff
@@ -0,0 +1,142 @@
+}
+
+type FFCClient struct {
+	restClient rest.Interface
+}
+
+func createFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(obj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("object is not a FeatureFlagConfiguration")
+	}
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeCreate,
+			},
+		}
+	}
+	return nil
+}
+
+func updateFuncHandler(oldObj interface{}, newObj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(oldObj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("old object is not a FeatureFlagConfiguration")
+	}
+	if reflect.TypeOf(newObj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("new object is not a FeatureFlagConfiguration")
+	}
+	if oldObj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		// Only update if there is an actual featureFlagSpec change
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeModify,
+			},
+		}
+	}
+	return nil
+}
+
+func deleteFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) error {
+	if reflect.TypeOf(obj) != reflect.TypeOf(&ffv1alpha1.FeatureFlagConfiguration{}) {
+		return errors.New("object is not a FeatureFlagConfiguration")
+	}
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeDelete,
+			},
+		}
+	}
+	return nil
+}
+
+func WatchResources(l log.Entry, clientSet FFCInterface, refreshTime time.Duration,
+	object client.ObjectKey, c chan<- sync.INotify,
+) {
+	ns := "*"
+	if object.Namespace != "" {
+		ns = object.Namespace
+	}
+	_, ffConfigController := cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo metav1.ListOptions) (result runtime.Object, err error) {
+				return clientSet.FeatureFlagConfigurations(ns).List(lo)
+			},
+			WatchFunc: func(lo metav1.ListOptions) (watch.Interface, error) {
+				return clientSet.FeatureFlagConfigurations(ns).Watch(lo)
+			},
+		},
+		&ffv1alpha1.FeatureFlagConfiguration{},
+		refreshTime,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				if err := createFuncHandler(obj, object, c); err != nil {
+					l.Warn(err.Error())
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				if err := deleteFuncHandler(obj, object, c); err != nil {
+					l.Warn(err.Error())
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				// This indicates a change to the custom resource
+				if err := updateFuncHandler(oldObj, newObj, object, c); err != nil {
+					l.Warn(err.Error())
+				}
+			},
+		},
+	)
+	go ffConfigController.Run(wait.NeverStop)
```

Comment:
I don't think this really matters, I did look at moving the errgroup down into this level but it involves modifying the interface. How else would you see this working? A local errgroup inside this function or something else @james-milligan ?

Reply:
I was thinking we could just pass through the context via `WatchResources` then pass the `ctx.Done` channel, again i'm not sure if its required, but if its offered as standard by the `Run` method it may be worth implementing

Reply:
Okay updated